src/tree/README.md

```diff
@@ -0,0 +1,285 @@
+# Custom components loaded in the tree
+
+## Intro
+The tree component in the Sitefinity AdminApp is used in various places in the application, for example, the grid (the list of content items), the sidebar, related data and so forth. This extension will allow you to inject a custom Angualr component in the tree's item template. The picture below shows where the custom component will be rendered in the tree's item template. 
+
+![Location of where the custom component will be rendered in the tree](./../../assets/tree/sample.png)
+
+### Demo for related data
+The custom component for the related data feature will be rendered, in the following parts of the UI
+* already selected related items
+* dropdown selector for the recent items and adjacent search feature
+* related data selector and adjacent search feature
+* readonly mode of the content item
+
+![Demo](./../../assets/tree/RelatedDataCustomTreeComponent.gif)
+
+The custom component you develop will get the currently rendered content item data passed into it, for example, the item passed in to the sample custom component has this json representation:
```

Comment:
aren't we passing a DataItem?

Reply:
yes we are, removing this entire section